### PR TITLE
$sample: Handle case of sample > collection / performance improvements

### PR DESCRIPTION
**Pull request by MatthiasEgli**
See commit messages for details

### CODE REVIEW

File: src/main/java/com/github/fakemongo/impl/aggregation/Sample.java [hunk 1]
```diff
@@ -4,19 +4,33 @@
...
+import java.util.Set;
 
 /**
  */
 @ThreadSafe
 public class Sample extends PipelineKeyword {
   public static final Sample INSTANCE = new Sample();
 
-  private static final Random RANDOM = new Random();
+  private static final Random rnd = new Random();
```
Comment (twillouer) [inline src/main/java/com/github/fakemongo/impl/aggregation/Sample.java]: why ?

File: src/test/java/com/github/fakemongo/FongoAggregateTest.java [hunk 2]
```diff
@@ -795,17 +795,29 @@ public void should_cursor_next_works() {
...
-    // Then
+    // Then return sampled collection
     List<DBObject> resultAggregate = Lists.newArrayList(cursor);
     Assertions.assertThat(resultAggregate).hasSize(3).doesNotContainNull();
     for (DBObject dbObject : resultAggregate) {
       Assertions.assertThat(dbObjects).contains(dbObject);
     }
+
+    // When sample size bigger collection size
+    BasicDBObject objBigSample = new BasicDBObject("$sample",
```
Comment (twillouer) [inline src/test/java/com/github/fakemongo/FongoAggregateTest.java]: can you provide an another test for this case ? (one for small, one for bigger)

**Comment by twillouer** [comment]
Thanks for your work !

File: src/main/java/com/github/fakemongo/impl/aggregation/Sample.java (hunk 1 above)
Comment (MatthiasEgli) [inline src/main/java/com/github/fakemongo/impl/aggregation/Sample.java]: No reason. Copied it from the stack overflow example, didn't change the name to your original one.

File: src/test/java/com/github/fakemongo/FongoAggregateTest.java (hunk 2 above)
Comment (MatthiasEgli) [inline src/test/java/com/github/fakemongo/FongoAggregateTest.java]: Sure, Will update my PR

File: src/main/java/com/github/fakemongo/impl/aggregation/Sample.java (hunk 1 above)
Comment (twillouer) [inline src/main/java/com/github/fakemongo/impl/aggregation/Sample.java]: I'm asking myself if he need to be uppercase or not..

http://stackoverflow.com/questions/1417190/should-a-static-final-logger-be-declared-in-upper-case

not sure..

**Comment by twillouer** [comment]
great, thanks !

**Pull request merged**